src/pages/settings/Settings.tsx

```diff
@@ -144,62 +138,39 @@ const Settings: FC = () => {
         configField.shortdesc?.toLowerCase().includes(query.toLowerCase())
       );
     })
-    .map((configField, index, { length }) => {
-      const isDefault = !Object.keys(settings?.config ?? {}).some(
-        (key) => key === configField.key,
+    .map((configField) => {
+      const clusteredValue = getConfigFieldClusteredValue(
+        clusteredSettings,
+        configField,
       );
-      const value = getValue(configField);
-
-      const clusteredValue = getClusteredValue(clusteredSettings, configField);
-
       const isNewCategory = lastCategory !== configField.category;
       lastCategory = configField.category;
-
-      return {
-        key: configField.key,
-        columns: [
-          {
-            content: isNewCategory && (
-              <h2 className="p-heading--5">{configField.category}</h2>
-            ),
-            role: "rowheader",
-            className: "group",
-            "aria-label": "Group",
-          },
-          {
-            content: (
-              <div className="key-cell">
-                {isDefault ? (
-                  configField.key
-                ) : (
-                  <strong>{configField.key}</strong>
-                )}
-                <p className="p-text--small u-text--muted u-no-margin--bottom">
-                  <ConfigFieldDescription description={configField.shortdesc} />
-                </p>
-              </div>
-            ),
-            role: "cell",
-            className: "key",
-            "aria-label": "Key",
-          },
-          {
-            content: (
-              <SettingForm
-                configField={configField}
-                value={value}
-                clusteredValue={clusteredValue}
-                isLast={index === length - 1}
-              />
-            ),
-            role: "cell",
-            "aria-label": "Value",
-            className: "u-vertical-align-middle",
-          },
-        ],
-      };
+      return getSettingRow(
+        configField,
+        isNewCategory,
+        clusteredValue,
+        deleteUserSetting,
+        settings,
+      );
     });
 
+  userSettings.forEach((s, index) => {
```

Comment:
We should also filter based on the query. Maybe extract the filter function as below and reuse it here and above:

```
  const matchesQuery = (configField: ConfigField) => {
    if (!query) {
      return true;
    }
    return (
      configField.key.toLowerCase().includes(query.toLowerCase()) ||
      configField.shortdesc?.toLowerCase().includes(query.toLowerCase())
    );
  };
  ```